packages/source-strapi/index.js

```diff
@@ -0,0 +1,77 @@
+const axios = require('axios')
+const { pascalCase } = require('change-case')
+const query = require('./helpers/query')
+
+module.exports = function (api, options) {
+  const coreProperties = [
+    'title',
+    'id',
+    'slug',
+    'path',
+    'date',
+    'content',
+    'excerpt'
+  ]
+
+  api.loadSource(async ({ addContentType, slugify }) => {
+    const { apiURL, queryLimit, contentTypes, loginData } = options
+    let jwtToken = null
+
+    // Check if loginData is set.
+    if (
+      loginData.hasOwnProperty('identifier') &&
+      loginData.identifier.length !== 0 &&
+      loginData.hasOwnProperty('password') &&
+      loginData.password.length !== 0
+    ) {
+      console.time('Authenticate Strapi user')
+      console.log('Authenticate Strapi user')
+
+      // Define API endpoint.
+      const loginEndpoint = `${apiURL}/auth/local`
+
+      // Make API request.
+      try {
+        const loginResponse = await axios.post(loginEndpoint, loginData)
+
+        if (loginResponse.hasOwnProperty('data')) {
+          jwtToken = loginResponse.data.jwt
+        }
+      } catch (e) {
+        console.error('Strapi authentication error: ' + e)
+      }
+
+      console.timeEnd('Authenticate Strapi user')
+    }
+
+    return Promise.all(contentTypes.map(resourceName => {
+      const typeName = pascalCase(resourceName)
```

Comment:
```suggestion
      const typeName = store.makeTypeName(resourceName)
```
The plugin should require a `typeName` option which defaults to `Strapi` instead of the hard-coded type name.